src/lib.rs

```diff
@@ -505,6 +526,49 @@ where
     }
 }
 
+impl<T, Q, N> Index<&'_ Q> for Counter<T, N>
+where
+    T: Hash + Eq + Borrow<Q>,
+    Q: Hash + Eq,
+    N: Zero,
+{
+    type Output = N;
+
+    /// Index in immutable contexts
+    ///
+    /// ```
+    /// # use counter::Counter;
+    /// let counter = Counter::<_>::init("aabbcc".chars());
+    /// assert_eq!(counter[&'a'], 2);
+    /// assert_eq!(counter[&'b'], 2);
+    /// assert_eq!(counter[&'c'], 2);
+    /// assert_eq!(counter[&'d'], 0);
+    /// ```
+    fn index(&self, key: &'_ Q) -> &N {
+        self.map.get(key).unwrap_or(&self.zero)
+    }
+}
+
+impl<T, N> IndexMut<&'_ T> for Counter<T, N>
+where
+    T: Clone + Hash + Eq,
+    N: Zero,
```

Comment:
Why no `Q` for `IndexMut`?

Reply:
To use `HashMap::entry` with `Clone::clone`. But I got a better solution. I modified the part to use `ToOwned` instead of `Clone`.